### PR TITLE
CI2, uniform 1.11 and bumped other dependencies

### PR DESCRIPTION
**Pull request by andynygard**
This is bringing maestro in line with the latest version of uniform and its other dependencies.

Also changing our CI to use ci2, to resolve the AFO issue.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,7 +30,9 @@ notifications:
...
 env:
   global:
-    secure: INxboouYQvs8qR05L6lg3ww3p63XkCuIryRoJSTsbJSybwvI4/Q+CoQRp3f5itiizjdSASjFIuE75//ZhRRzcXYTDwg+ohY1j6IjL/srrfuQE57tUtlmbJDcicktQwMZ4hCSuNBWTeUCzsECcZXWwQ7A0NY7UssZH1FSDDVitxA=
+  - CI_VERSION=3.2.2-20160718231250-50a210b
+  - secure: qmguQnnRSSLTFlnBleczJ8yn0WUM+mRFeDACwyH5dB0RIxdFOWOw5ai292zCL8m+zuD8On7D1l09P3lwKq0ubvaCMdH0u3YufJo12xzl6Hzh0oNeXIgxca8DuBQIP865zP0J+9Ksav3Zc9QNqM5aLk/S7RV2ejRrTwJRnYbXOt8=
+  - secure: binZQiassnDLS2sVesj1dHId1LjVxBQxFdAO817pgByvxFXM0egpnEmneVW08bqh+FcBeSFMVIfxgDzenwu0QfUv1BbfqnHfbq/ou9jJy9KL6dIWYqXQU1FoPfVT37vtOvQ9gzl80gjez53Ay2hJyzW6VILeiN1HpnoBp20W05k=
 before_install:
 - openssl aes-256-cbc -K $encrypted_680e980041ef_key -iv $encrypted_680e980041ef_iv
-  -in .travis/deploy-key.enc -out .travis/deploy-key.pem -d
+  -in .ci/deploy-key.enc -out .ci/deploy-key.pem -d
```
Comment (vineethvarghese) [inline .travis.yml]: Did you add this?

Comment (andynygard) [inline .travis.yml]: Yep, the key now lives in `.ci` so that ci2 doesn't need to care about which CI tool we're using.

File: project/build.scala [hunk 2]
```diff
@@ -34,14 +34,14 @@ import au.com.cba.omnia.humbug.HumbugSBT._
...
-  val humbugVersion      = "0.7.1-20160117233334-0be7ee9"
-  val parlourVersion     = "1.11.2-20160118033414-b9324c5"
+  val thermometerVersion = "1.4.3-20160617114144-562e6e0"
+  val ebenezerVersion    = "0.22.2-20160619063420-4eb964f"
+  val beeswaxVersion     = "0.1.2-20160619053150-80dbb0a"
+  val omnitoolVersion    = "1.14.1-20160617114243-1143bc9"
+  val permafrostVersion  = "0.13.0-20160718235343-68e0f07"
+  val edgeVersion        = "3.7.0-20160719010243-5c6c2dd"
+  val humbugVersion      = "0.7.2-20160617114039-c10e869"
+  val parlourVersion     = "1.12.1-20160719055041-8a79eeb"
```
Comment (rowandavies) [inline project/build.scala]: Nice, I was just considering bumping these versions, starting with omnitool, but for now had turned off strict dependency management in the etl test job.

File: .travis.yml [hunk 3]
```diff
@@ -7,13 +7,17 @@ cache:
...
+- cp /etc/sbt/sbtopts .sbtopts; echo "-Dsbt.global.base=$TRAVIS_BUILD_DIR/ci" >> .sbtopts
 script:
-- sbt -Dsbt.global.base=$TRAVIS_BUILD_DIR/ci '; test; package; project benchmark;
-  test:compile; project example; test; assembly' && ci/sbt-deploy.sh && ci/sbt-gh-pages-ssh.sh
+- sbt test package
+  && sbt "project benchmark" test:compile
+  && sbt "project example" test assembly
+  && ci/sbt-ci-deploy.sh maven http://commbank.artifactoryonline.com/commbank ext-releases-local
+  && ci/sbt-ci-build-doc.sh https://commbank.github.io/ https://github.com/CommBank/maestro/
+  && ci/ci-push-branch.sh gh-pages
```
Comment (rowandavies) [inline .travis.yml]: Just to check: is there a specific reason to change this to run as separate sbt commands?

**Comment by rowandavies** [comment]
LGTM  👍

File: .travis.yml (hunk 3 above)
Comment (andynygard) [inline .travis.yml]: I think it makes it more readable, as at first I was wondering why we were calling `test` twice, whereas with each project on one line I think it's a bit clearer.

**Comment by andynygard** [comment]
As discussed, I've updated the line to:
`sbt test package benchmark/test:compile example/test example/assembly`

which should satisfy both concerns!

**Comment by vineethvarghese** [comment]
👍

**Pull request merged**